routingapi/api.go

```diff
@@ -69,14 +69,22 @@ func (r *RoutingAPI) makeTcpRouteMapping(route config.Route) (models.TcpRouteMap
 
 	r.logger.Info("Creating mapping", lager.Data{})
 
-	return models.NewTcpRouteMapping(
+	return models.NewSniTcpRouteMapping(
 		routerGroupGUID,
 		uint16(*route.ExternalPort),
+		nilIfEmpty(&route.ServerCertDomainSAN),
```

Comment:
`route.ServerCertDomainSAN` type definition is:

`ServerCertDomainSAN  string`

therefore, it can never be `nil`. 

Suggestion: 
```
func nilIfEmpty(str string) *string {
	if *str == "" {
		return nil
	}
	return str
}
```

Reply:
I think you mean:
```go
func nilIfEmpty(str string) *string {
	if str == "" {
		return nil
	}
	return &str
}
```
but this relies on copies of string values... I think, not so good with Go, better with C++. Taking the address of the string in the param avoids the copy.